pkg/scheduler/core/spreadconstraint/select_clusters_by_region_test.go

```diff
@@ -160,6 +160,235 @@ func Test_selectBestClustersByRegion(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "select cluster from the region with enough clusters and higher score",
+			args: args{
+				spreadConstraintMap: map[policyv1alpha1.SpreadFieldValue]policyv1alpha1.SpreadConstraint{
+					policyv1alpha1.SpreadByFieldRegion: {
+						SpreadByField: policyv1alpha1.SpreadByFieldRegion,
+						MaxGroups:     2,
+						MinGroups:     2,
+					},
+					policyv1alpha1.SpreadByFieldCluster: {
+						SpreadByField: policyv1alpha1.SpreadByFieldCluster,
+						MaxGroups:     5,
+						MinGroups:     4,
+					},
+				},
+				groupClustersInfo: &GroupClustersInfo{
+					Regions: map[string]RegionInfo{
+						"R1": {
+							Name:              "R1",
+							Score:             90,
+							AvailableReplicas: 40,
+							Clusters: []ClusterDetailInfo{
+								{
+									Name:              "member1",
+									Score:             90,
+									AvailableReplicas: 40,
+									Cluster:           NewClusterWithTopology("member1", "P1", "R1", "Z1"),
```

Comment:
sum of clusters' AvailableReplicas should be same with region group's AvailableReplicas

Reply:
Yeah, but I don't see any problem.

Reply:
Just for making people not confused when they see the ut.

Reply:
Sorry, you already fixed it in your newest commit.

Reply:
for the case `no enough region`, `R1` has two clusters(40/40 AvailableReplicas), but this group has 100 AvailableReplicas, can you help fix the related problem?

Reply:
I get it. I thought those test cases that I recently added had this problem. I'll fix them together in my next commit.